src/taskpane/server.js

```diff
@@ -2237,7 +2249,7 @@ async function sendToSpira(model, fieldTypeEnums, isUpdate) {
   } else {
     return await Excel.run({ delayForCellEdit: true }, function (context) {
       var sheet = context.workbook.worksheets.getActiveWorksheet(),
-        sheetRange = sheet.getRangeByIndexes(1, 0, MAX_ROWS_PER_PAGE, fields.length);
+        sheetRange = sheet.getRangeByIndexes(1, 0, MAX_ROWS_PER_PAGE_MAX, fields.length);

```

Comment:
Why is this using the MAX_ROWS_PER_PAGE_MAX var and not the dynamic MAX_ROWS_PER_PAGE?
I think this is for loading the template, but not sure why it will always be 2000 rows?

Reply:
Because here we are not loading any data from the API, but rather sending data to Spira. Excel requires us to provide a range number for the page. Ideally, this should be "send all the data in this spreadsheet" - since we don't have that infinite row option, we stick with the max 2,000 rows limit. Does it make sense?

Reply:
So here we are finding the rows to send to Spira from Excel?

1. What do we do for eg Incidents if people have added more than 2000 rows of data?
2. For Test cases with test steps, they may have entered a lot less than 200 test cases in 2000 rows. So how should we handle that?

Basically, this approach may cut off what data is uploaded to Excel and users will not know. The easiest answer is to select a much bigger number I think. Maybe 64000? Unlike Sheets, Excel always has a lot of rows so we should be safe in determining that but should check on web and desktop

Reply:
That makes sense. I modified it to look for 64k rows when sending. I believe this does not impact performance, as it stops scanning when it detects an empty row.